### PR TITLE
Initial commit for extended python SDK, still WIP

### PR DESCRIPTION
**Pull request by stephan271**
To get initial feedback

### CODE REVIEW

File: demo/demoTest2.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#logbooks = log.get_logbooks(ownerGroup=pgroup)
+
+# most flexible case: specify the where expression in Mongo Syntax as python dictionary
+
+logbooks = log.get_logbooks(where={"ownerGroup": pgroup},limit=10,skip=0)
+assert len(logbooks) >= 1
+logbook = logbooks[0]
+# print("Logbook:",logbook)
+
+# make sure that further actions to this logbook
```
Comment (wakonig) [inline demo/demoTest2.py:40]: maybe "make sure that further actions will be pinned to this logbook"? or "will be assigned..."?

File: sdk/python/scilog/scilog.py [hunk 2]
```diff
@@ -209,10 +245,10 @@ def append_files_to_snippet(self, snippet: Paragraph, filepaths: list, **kwargs)
         return self.patch_snippet(snippet)
 
     @typechecked
+    # TODO : User function ? Add docstring
     def send_logbook_message(self, msg: lm.LogbookMessage) -> None:
         payload = msg._content.to_dict(include_none=False)
         payload["linkType"] = "paragraph"
-        # FIXME the python sdk should not use the ownergroup
```
Comment (wakonig) [inline sdk/python/scilog/scilog.py:215]: I think the problem still remains.

File: sdk/python/scilog/snippet.py [hunk 3]
```diff
@@ -16,12 +16,11 @@ def typechecked_call(obj, *args, **kwargs):
                 )
             if len(func.__closure__) > 0:
                 property_name = func.__closure__[0].cell_contents.strip("_")
-                if obj._deprecated.get(property_name):
-                    warnings.warn(
-                        f"{property_name} is deprecated by {obj._deprecated_by[property_name]}"
-                    )
+                # if obj._deprecated.get(property_name):
```
Comment (wakonig) [inline sdk/python/scilog/snippet.py:19]: why did you remove the deprecation? If this was on purpose, we could remove the warning "catcher" from send_logbook_message.

File: demo/demoTest2.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+log.select_logbook(logbook)
+
+# example of *adding* information (text,images,tags) to this  logbook
+# TODO: currently gives error: {"files":["is not defined in the model"],"linkType":["is not defined in the model"]}
+# msg = LogbookMessage()
```
Comment (wakonig) [inline demo/demoTest2.py:45]: I'm a bit puzzled by this. Where does the error occur? In logbook_message.py? Internally, the LogbookMessage class uses a Paragraph, so it should have files and linkType.

File: demo/demoTest2.py (hunk 1 above)
Comment (stephan271) [reply demo/demoTest2.py]: done

File: sdk/python/scilog/snippet.py (hunk 3 above)
Comment (stephan271) [reply sdk/python/scilog/snippet.py:19]: Reenabled

File: sdk/python/scilog/scilog.py (hunk 2 above)
Comment (stephan271) [reply sdk/python/scilog/scilog.py:215]: Solved

File: docs/Ingestor/PythonSDK.md [hunk 5]
```diff
@@ -1,9 +1,84 @@
 # Use of Python SDK
 
-You find an example script in the demo folder https://github.com/paulscherrerinstitute/scilog/tree/main/demo . The example will add a paragraph text and a second paragraph including two figures.
+You find an example script in the demo folder https://github.com/paulscherrerinstitute/scilog/tree/main/demo/demo.py . The example will add a paragraph containg tet, images and tags as well querying the existing snippet data in a logbook.
```
Comment (minottic) [inline docs/Ingestor/PythonSDK.md:3]: there's a typo here `tet` instead of `test`

File: sdk/python/scilog/snippet.py (hunk 3 above)
Comment (minottic) [reply sdk/python/scilog/snippet.py:19]: I think instead of commenting this, we could use logger.log(...) instead of warning.warn so that it is less intrusive. I'd keep a log as it is indeed something that the python users want to be informed

File: sdk/python/scilog/httpclient.py [hunk 6]
```diff
@@ -96,34 +104,12 @@ def patch_request(self, url, payload=None, files=None, headers=None, timeout=10)
...
-        if skip > 0:
-            filt["skip"] = skip
-        if fields is not None:
-            filt["fields"] = include
-        if order is not None:
-            filt["order"] = order
-        filt = json.dumps(filt)
-        return {"filter": filt}
+        )
+        # print("Result of login request:",x)
```
Comment (minottic) [inline sdk/python/scilog/httpclient.py:114]: maybe remove?

File: sdk/python/scilog/scilog.py [hunk 7]
```diff
@@ -12,27 +13,39 @@
...
 
 
+logger = logging.getLogger(__name__)
+
+
 def pinned_to_logbook(logbook_keys, include_none=False):
     def pinned_to_logbook_inner(func):
         @functools.wraps(func)
-        def pinned_to_logbook_call(log, *args, **kwargs):
+        def pinned_to_logbook_call(log, *args, **kwargs):  # vor AUfruf
```
Comment (minottic) [inline sdk/python/scilog/scilog.py:27]: maybe remove this :"# vor AUfruf"

**Review by minottic — COMMENTED** [review]
Some picky comments, apart from the typo and the one about the deprecation warning, which I would suggest changing. If not, we can always do in a later PR. I am happy to approve otherwise

File: sdk/python/scilog/snippet.py (hunk 3 above)
Comment (minottic) [reply sdk/python/scilog/snippet.py:19]: or logger.warning:
https://stackoverflow.com/questions/9595009/warnings-warn-vs-logging-warning

Comment (stephan271) [reply sdk/python/scilog/snippet.py:19]: LGTM, logging instead of warning

File: docs/Ingestor/PythonSDK.md (hunk 5 above)
Comment (stephan271) [reply docs/Ingestor/PythonSDK.md:3]: Thanks for fixing

File: sdk/python/scilog/httpclient.py (hunk 6 above)
Comment (stephan271) [reply sdk/python/scilog/httpclient.py:114]: Yes

**Review by minottic — APPROVED** [review]
